### ISSUE TITLE
Option to use Postmark's templates API

### ISSUE THREAD

**Issue by MatanYadaev** [coconutcraig/laravel-postmark#28]
Instead of only using Laravel's mail builder schema, this package should have an option to use Postmark's templates API, which is one of the main features of Postmark.

https://postmarkapp.com/developer/api/templates-api

**Comment by craigpaul** [comment]
Hey @MatanYadaev, I think that sounds like a good idea. How would you imagine that works within a Laravel application?

**Comment by MatanYadaev** [comment]
Hi Craig, thanks! 😃

I imagine it will look like this:

```php
/**
 * Get the mail representation of the notification.
 *
 * @param  mixed  $notifiable
 * @return \Illuminate\Notifications\Messages\MailMessage
 */
public function toMail($notifiable)
{
    $templateId = '123456789';

    return (new MailMessage)->template($templateId, [
        'invoice' => $this->invoice
    ]);
}
```

I am not sure it's possible to implement it this way. I don't really know how Laravel handles mail notifications, so sorry if this implementation looks weird :)

Anyway, it sounds like an important feature for me, I am actually using my own Postmark class because of 
this reason, and I'm not using Laravel's Notification class.
I would like to use Laravel's Notification class and this package for mails, if this feature will be supported.

EDIT:
BTW, I will be glad to contribute and create the PR for this feature request, just let's discuss about the design.

**Comment by craigpaul** [comment]
Hey @MatanYadaev, I think it could be a good feature. I've personally never used Postmark's templates, but I'm totally cool with this package supporting them. I think it would have to be an extension of Laravel's MailMessage class, but it already has a template method, so we should come up with a different name I'm thinking. According to the documentation, the items that this would need to support (that aren't already supported) are `TemplateId` (or `TemplateAlias`) and `TemplateModel`.

I'd gladly accept a PR for this and your implementation looks good to me, we'd just need a different naming structure as that method name already exists and then any amended handling of the sending through the transport. If you're unable to work on this, I'll give it a try, but I can't promise any timelines at this moment. Let me know what you think.

**Comment by MatanYadaev** [comment]
@craigpaul I actually don't really know how Laravel's Notification & Mail modules work, but I really want this feature to be included in this package, so I will give it a try, I will begin to read and research about the Notification & Mail modules.
If you have any tips, short intro, or how do you think this feature should be implemented - it would be nice and helpful.

**Comment by craigpaul** [comment]
@MatanYadaev That's fair, I have some time this weekend to take a look into how I would like to see this available. If its not too complicated I will throw something together and put up a PR 👍

**Comment by MatanYadaev** [comment]
@craigpaul I actually began to write some code, I want to discuss with you about how you would implement it:
(let's ignore the naming convention for now, we'll cover it later)
I created a new class called `MailTemplateMessage` that inherits Laravel's `MailMessage`.
This class has some methods to define `TemplateId`, `TemplateAlias`, and `TemplateModel`.
How should I identify the incoming message in the `PostmarkTransport@send` method as a `MailTemplateMessage` message?
I see the message that passes in the first argument is an instance of `Swift_Mime_SimpleMessage` and not a `MailMessage`.
Any suggestions?

**Comment by craigpaul** [comment]
Hey @MatanYadaev, sorry for the delay. I just tried getting a **very rough** implementation of this going. You can [check it out here](https://github.com/craigpaul/laravel-postmark/commit/69ac01ba4c54d7ec499253c3e461c7d443468145). This approach utilizes a new `MailMessage` class that ends up rendering a view that returns a stringified JSON object. That string gets set as the body property on the `Swift_Mime_SimpleMessage` class that is passed to the send method. Hopefully this gives you some direction on how to get that information to the transport. 

Note: I'm not saying it has to be done this way, this is just the way I found to get it in there 👍 

Let me know if that helps

**Comment by MatanYadaev** [comment]
@craigpaul Looks like a good implementation. It took me some time to understand this `Swift` stuff and how it works behind the scenes.
Well, let's just finish this feature, do you want me to help you with this?

**Comment by craigpaul** [comment]
This is now available in v2.5.0, thanks for the feature request 😄
